### ISSUE TITLE
Why "carouselCurrentItemIndexDidChange" method is calling twice when reloadData is active?

### ISSUE THREAD

**Issue by goktugaral** [nicklockwood/iCarousel#686]
Hi there,

First of all thanks for this valuable lib. 

I want to ask you what is the reason of "carouselCurrentItemIndexDidChange" method is calling twice when reloadData is call?

Basically, this method exactly what I'm looking for.  When my view its starts to change, I can get correct currentItemIndex and then I'm going to do some work while user is swipe. But this "calling twice" thing is break my logic. 

I have to deal with this method only if user swipes the views not with the reloadData calls. I temporary added a flag which is controlling is it calling twice at the beginning or not. 
I'm wondering if there is a better solution or what is the reason of it?

Thanks for your advice! Cheers!

**Comment by UHemarajU** [comment]
Hi goktugaral,

Actually it executes only once. May be your are written this reload in two methods. They are carouselCurrentItemIndexDidChange and didSelectItemAtIndex. Please check once.

**Comment by goktugaral** [comment]
Thank you for your respond. I've checked and I call "reloadData" method only one time. 
Yes there are two (taps) method and I'm using carouselCurrentItemIndexDidChange. didSelectItemAtIndex also include but inside of the method is empty. 

By the way, "carouselCurrentItemIndexDidChange" method is calling even before "reloadData" method. After "reloadData" is active it triggers again.

**Comment by goktugaral** [comment]
Hi again! I think, I found a reason for what might have been done. I set my view's class as iCarousel on Identity Inspector and it was initializing the via nib. @UHemarajU you were right. There is reloadData method in setDataSource method. Before my logic starts (I don't have any array for numberOfItemsInCarousel at this moment), this setDataSource method was calling and the process calls the "carouselCurrentItemIndexDidChange". I was trying to avoid that.

Thanks for your help anyway.
